### PR TITLE
Phocube add all bands Closes #3877

### LINKED ISSUE #3877: phocube, in source=projection should be able to accept a multi-band cube as input
**Description**  

Adding lat/lon backplanes to a color mosaic would be easier if phocube would accept a multiband input.  I think it is reasonable to assume that if the user wants to propagate DNs to the output, they will want to propagate DNs for all the bands. Yes, one can run phocube on a single band and then run cubeit, but, for large mosaics, that adds another slow step.  A nice-to-have, not something essential! 
**Example**  

enter the following command:
_phocube from=some-color-mosaic.cub to=mosaic-with-backplanes.cub source=projection_
and you get an output instead on an error message.

### PR DESCRIPTION
**Pull request by scsides**
New option to phocube to transfer all the input bands to the output file.

## Description

The phocube program is now able to transfer all of the input bands to the output cube using the parameter "ALLDNS". Previously it required a single band cube as input and the "DNS" parameter would cause the first output band to be the DNs from the input cube. 

## Related Issue
#3877

## Motivation and Context
User requested option

## How Has This Been Tested?
Hand tests were run and a gtest was created.

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [X] New feature (non-breaking change which adds functionality)
- [ ] Documentation change (update to the documentation; no code change)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [ ] I have read and agree to abide by the [Code of Conduct](https://github.com/USGS-Astrogeology/ISIS3/blob/dev/Code-Of-Conduct.md)
- [X] I have read the [**CONTRIBUTING**](https://github.com/USGS-Astrogeology/ISIS3/blob/dev/CONTRIBUTING.md) document.
- [X] My change requires a change to the documentation.
- [X] I have updated the documentation accordingly.
- [X] I have added tests to cover my changes.
- [X] All new and existing tests passed.
- [X] I have added myself to the [.zenodo.json](https://github.com/USGS-Astrogeology/ISIS3/blob/dev/.zenodo.json) document.
- [X] I have added any user impacting changes to the [CHANGELOG.md](https://github.com/USGS-Astrogeology/ISIS3/blob/dev/CHANGELOG.md) document.

## Licensing
This project is mostly composed of free and unencumbered software released into the public domain, and we are unlikely to accept contributions that are not also released into the public domain. Somewhere near the top of each file should have these words:

> This work is free and unencumbered software released into the public domain. In jurisdictions that recognize copyright laws, the author or authors of this software dedicate any and all copyright interest in the software to the public domain.

- [X] I dedicate any and all copyright interest in this software to the public domain. I make this dedication for the benefit of the public at large and to the detriment of my heirs and successors. I intend this dedication to be an overt act of relinquishment in perpetuity of all present and future rights to this software under copyright law.

### CODE REVIEW

File: isis/tests/FunctionalTestsPhocube.cpp [hunk 1]
```diff
@@ -346,4 +353,38 @@ TEST_F(DefaultCube, FunctionalTestPhocubeNoBandBin) {
...
+  ASSERT_EQ(cube.sampleCount(), testCube->sampleCount());
+  ASSERT_EQ(cube.lineCount(), testCube->lineCount());
+  ASSERT_EQ(cube.bandCount(), 8);
+
+  PvlGroup bandBin = isisLabel->findGroup("BandBin", Pvl::Traverse);
+  EXPECT_PRED_FORMAT2(AssertQStringsEqual, bandBin.findKeyword("FilterName")[0], "B1");
+  EXPECT_PRED_FORMAT2(AssertQStringsEqual, bandBin.findKeyword("FilterName")[1], "B2");
+  EXPECT_PRED_FORMAT2(AssertQStringsEqual, bandBin.findKeyword("FilterName")[2], "B3");
+
+  std::unique_ptr<Histogram> hist (cube.histogram(0));
```
Comment (jessemapel) [inline isis/tests/FunctionalTestsPhocube.cpp]: This should check all 3 bands

File: isis/tests/FunctionalTestsPhocube.cpp [hunk 2]
```diff
@@ -346,4 +353,38 @@ TEST_F(DefaultCube, FunctionalTestPhocubeNoBandBin) {
...
+  EXPECT_NEAR(hist->Average(), 67.575204, .000001);
+  EXPECT_NEAR(hist->Sum(), 13515.040821, .000001);
+  EXPECT_EQ(hist->ValidPixels(), 200);
+  EXPECT_NEAR(hist->StandardDeviation(), 76.089184, .000001);
```
Comment (jessemapel) [inline isis/tests/FunctionalTestsPhocube.cpp]: Instead of hard coded values, pull the histogram from the input cube and its values for comparison.

**Review by jessemapel — COMMENTED** [review]
Logic looks good. Needs a changelog entry and the test could be more robust

File: isis/tests/FunctionalTestsPhocube.cpp (hunk 1 above)
Comment (scsides) [reply isis/tests/FunctionalTestsPhocube.cpp]: done

File: isis/tests/FunctionalTestsPhocube.cpp (hunk 2 above)
Comment (scsides) [reply isis/tests/FunctionalTestsPhocube.cpp]: done

**Review by jessemapel — APPROVED** [review]
Looks good now

**Pull request merged**